src/components/screens/all-software-screen.tsx

```diff
@@ -0,0 +1,505 @@
+
+      if (appToUpdate) {
+        const updatedApp: SoftwareRegistry = {
+          ...appToUpdate,
+          status: newStatus,
+          name: appToUpdate.name || "",
+          description: appToUpdate.description || "",
+          review: appToUpdate.review || "",
+          url: appToUpdate.url || "",
+          image: appToUpdate.image || "",
+          tags: appToUpdate.tags || [],
+          users: appToUpdate.users || []
+        };
+
+        const updatedApplications = applications.map(app =>
+          app.id === id ? updatedApp : app
+        );
+        setApplications(updatedApplications);
+
+        await softwareApi.updateSoftwareById({
+          id,
+          softwareRegistry: updatedApp
+        });
+      }
+    } catch (error) {
+      console.error(`Error updating status: ${error}`);
+    }
+  };
+
+  /**
+   * Adds the current user to the list of users for the specified application.
+   * 
+   * @param {string} id - The id of the application to save.
+   */
+  const handleSave = async (id: string) => {
+    try {
+      const applicationToUpdate = applications.find(app => app.id === id);
+      if (!applicationToUpdate) {
+        throw new Error(`Application with ID ${id} not found`);
+      }
+
+      const updatedUsers = applicationToUpdate.users ? [...applicationToUpdate.users, loggedUserId] : [loggedUserId];
+
+      const updatedApplications = applications.map(app =>
+        app.id === id ? { ...app, users: updatedUsers, isInMyApplications: true } : app
+      );
+      setApplications(updatedApplications);
+
+      await softwareApi.updateSoftwareById({
+        id,
+        softwareRegistry: {
+          ...applicationToUpdate,
+          users: updatedUsers,
+        }
+      });
+
+    } catch (error) {
+      console.error(`Error saving the app: ${error}`);
+    }
+  };
+
+  /**
+   * Opens the delete confirmation dialog.
+   *
+   * @param id - The ID of the application to delete.
+   */
+  const openDeleteDialog = (id: string) => {
+    setSelectedApplicationId(id);
+    setDeleteDialogOpen(true);
+  };
+
+  /**
+   * Closes the delete confirmation dialog.
+   */
+  const closeDeleteDialog = () => {
+    setSelectedApplicationId(null);
+    setDeleteDialogOpen(false);
+  };
+
+  /**
+   * Handles the removal of an application after confirmation.
+   */
+  const handleRemove = async () => {
+    if (!selectedApplicationId) return;
+
+    try {
+      const applicationToDelete = applications.find(app => app.id === selectedApplicationId);
+      if (!applicationToDelete) {
+        throw new Error(`Application with ID ${selectedApplicationId} not found`);
+      }
+
+      const updatedApplications = applications.filter(app => app.id !== selectedApplicationId);
+      setApplications(updatedApplications);
+
+      await softwareApi.deleteSoftwareById({ id: selectedApplicationId });
+    } catch (error) {
+      console.error(`Error deleting the app: ${error}`);
```

Comment:
Error handler, I won't leave any more comments about this, check for use of console.error that it would be appropriate to replace with the error handler e.g. setError (from atom)